src/main/java/de/retest/recheck/RecheckImpl.java

```diff
@@ -146,7 +146,9 @@ protected ActionReplayResult createActionReplayResult( final Object toVerify, fi
 		final RecheckDifferenceFinder finder =
 				new RecheckDifferenceFinder( adapter.getDefaultValueFinder(), currentStep, file.getPath() );
 
-		return finder.findDifferences( actual, expected );
+		final ActionReplayResult actionReplayResult = finder.findDifferences( actual, expected );
+		adapter.notifyAboutDifferences( actionReplayResult );
```

Comment:
Shouldn't this only happen when there are actually differences?

```suggestion
                if ( actionReplayResult.hasDifferences() ) {
                	adapter.notifyAboutDifferences( actionReplayResult );
                }
```